Add global flag --output_json, add class Display.

zofiakochutek opened the pull request.

zofiakochutek committed:
Add global flag --output_json, add class Display.

mkochanowski committed:
Merge branch 'develop' into issue-53-display-class

mkochanowski commented on client/maisie/cli/display.py:
```diff
@@ -0,0 +1,38 @@
+from colorama import Fore
+
+
+class Display:
```
The `Display` class should be included in the `utils` module. `cli` is meant to contain only method groups and methods executed directly in the CLI, where `Display` is just a helper class used for managing the formatting of the response.

mkochanowski commented on client/maisie/cli/users.py:
```diff
@@ -42,21 +38,20 @@ def rm():
+        title = "List of users"
+        include = DEFAULT_DISPLAY_ATTRIBUTES
+        display = Display(
+            context=context, response=users, attributes=include, title=title
```
I think that instead of creating an additional variable called `include` we can now pass the `DEFAULT_DISPLAY_ATTRIBUTES` itself directly to the Display method, using the supported `attributes` parameter.

mkochanowski commented on client/maisie/interface.py:
```diff
@@ -10,8 +10,15 @@
-def cli():
-    pass
+@click.option(
+    "--output_json", is_flag=True, help="Display output in pure json instead of a table"
```
I am still wondering if this is the right approach. What would happen if we wanted to add an additional way of formatting the response in the future? What would happen if I were to include both `--output_json` and `--output_xml` in the same time?

What if instead of creating multiple global options for formatting, we stick to the regular approach, meaning having a single option such as `--output=json`? It seems like that would easily allow us the extend the functionality when needed.
Your input on this issue would be most welcome.

mkochanowski commented on client/maisie/interface.py:
```diff
@@ -10,8 +10,15 @@
+    context.obj = {}
+    context.obj["output_json"] = False
+    if output_json:
+        context.obj["output_json"] = True
```
If I am not mistaken, this can also be written as:
```python
context.obj["output_json"] = output_json
```

zofiakochutek committed:
Requested changes

-> alter place of Display class to utils module
-> setting the --output to be an option instead of a flag
-> better the logic of interface

zofiakochutek committed:
Merge branch 'issue-53-display-class' of github.com:nokia-wroclaw/innovativeproject-ml-models-management into issue-53-display-class

mkochanowski commented on client/maisie/__init__.py:
```diff
@@ -53,3 +53,4 @@
 from maisie.resources.projects import Projects
 from maisie.resources.workspaces import Workspaces
 from maisie.resources.users import Users
+from maisie.utils.display import Display
```
Adding the import statement to this file will make the `Display` class available to all users of our Maisie package, alongside the `Projects`, `Workspaces`, `Users` classes designed to manage our Maisie instance. 

`Display` is just a helper class that is supposed to be used internally for formatting the responses of other methods, making it explicitly available to users should be discouraged.

mkochanowski commented on client/maisie/utils/display.py:
```diff
@@ -33,6 +33,6 @@ def create_table(self):
-        if not self.context.obj["output_json"]:
+        if not self.context.obj["output"] == "json":
             content_to_display = self.create_table()
         click.echo(content_to_display)
```
Great!

mkochanowski commented on client/maisie/interface.py:
```diff
@@ -10,15 +10,11 @@
-@click.option(
-    "--output_json", is_flag=True, help="Display output in pure json instead of a table"
-)
+@click.option("--output", help="Specify format of the output as e.g. json")
```
Maybe we should add a default value for this parameter, for example `--output=default` or `--output=table`?

zofiakochutek committed:
Set rightly import of class Display

zofiakochutek committed:
Changes relative imports to absolute
